src/Compilers/Core/Portable/Compilation/IExpression.cs

```diff
@@ -828,6 +829,80 @@ public static BinaryOperandsKind GetBinaryOperandsKind(BinaryOperationKind kind)
         }
     }
 
+    public interface IConversion
```

Comment:
I'm not sure that this interface is necessary. Most, if not all, of its properties can be implemented--however awkwardly--as extension methods over IConversionExpression. The ones that cannot be implemented efficiently as extension methods can inspire improvements to IConversionExpression.

I understand that the introduction of the interface is a reaction to earlier review comments, but another iteration on life without it might be worthwhile.